dm/syncer/data_validator.go

```diff
@@ -422,23 +534,77 @@ func (v *DataValidator) startValidateWorkers() {
 			worker.run()
 		}()
 	}
+
+	for _, tblChange := range v.loadedPendingChanges {
+		for key, row := range tblChange.rows {
+			v.dispatchRowChange(key, row)
+		}
+	}
 }
 
 func (v *DataValidator) dispatchRowChange(key string, row *rowChange) {
 	hashVal := int(utils.GenHashKey(key)) % v.workerCnt
 	v.workers[hashVal].rowChangeCh <- row
 }
 
+func (v *DataValidator) genValidateTableInfo(sourceTable *filter.Table) (*validateTableInfo, error) {
+	targetTable := v.syncer.route(sourceTable)
+	// todo: syncer will change schema in schemaTracker, will there be data race?
```

Comment:
tracker is thread-safe  but this may cause different table schema between syncer and validator in extreme cases

Reply:
validator is assuming user hasn't do any destructive ddl such as remove any existing column(in real production database, there should‘t be much delete columns operation in most case), so it's ok to use `current ddl` in syncer to validate previous binlog(it's corresponding  ddl may different) as long as the ddls is validation compatible.

@lance6716 says CDC is storing ddl history, we may reuse it, so validator can use ddl corresponding to binlog, but we may add it in later sprint.